lib/views/service/tabs/bill_of_lading.dart

```diff
@@ -0,0 +1,175 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:path_provider/path_provider.dart' as path_provider;
+import 'package:path_provider_platform_interface/path_provider_platform_interface.dart';
+// ignore: directives_ordering
+import 'package:open_file/open_file.dart' as open_file;
+// import 'package:native_pdf_view/native_pdf_view.dart';
+import 'package:syncfusion_flutter_pdf/pdf.dart';
+import 'package:syncfusion_flutter_pdfviewer/pdfviewer.dart';
+import 'package:transportation_mobile_app/models/entities/address.dart';
+import 'package:transportation_mobile_app/models/entities/globals.dart';
+import 'package:transportation_mobile_app/models/entities/inspection_item.dart';
+import 'package:transportation_mobile_app/models/entities/session.dart';
+
+class BillOfLadingPage extends StatelessWidget {
+
+  SessionObject session = getCurrentSession();
+
+  Future<File> saveAndLaunchFile(List<int> bytes, String fileName) async {
+    //Get the storage folder location using path_provider package.
+    String path;
+    if (Platform.isAndroid ||
+        Platform.isIOS ||
+        Platform.isLinux ||
+        Platform.isWindows) {
+      final Directory directory =
+      await path_provider.getApplicationSupportDirectory();
+      path = directory.path;
+    } else {
+      path = await PathProviderPlatform.instance.getApplicationSupportPath();
+    }
+    final File file =
+    File(Platform.isWindows ? '$path\\$fileName' : '$path/$fileName');
+    await file.writeAsBytes(bytes, flush: true);
+    if (Platform.isAndroid || Platform.isIOS) {
+      //Launch the file (used open_file package)
+      await open_file.OpenFile.open('$path/$fileName');
+    } else if (Platform.isWindows) {
+      await Process.run('start', <String>['$path\\$fileName'], runInShell: true);
+    } else if (Platform.isMacOS) {
+      await Process.run('open', <String>['$path/$fileName'], runInShell: true);
+    } else if (Platform.isLinux) {
+      await Process.run('xdg-open', <String>['$path/$fileName'],
+          runInShell: true);
+    }
+    return file;
+  }
+
+  void fillPDF(final bill) async {
+    final byteData = await rootBundle.load('assets/${bill}.pdf');
+    String templatePath = '${(await getTemporaryDirectory()).path}/${bill}_template.pdf';
+    final file = File(templatePath);
+    await file.writeAsBytes(byteData.buffer.asUint8List(byteData.offsetInBytes, byteData.lengthInBytes));
+    final PdfDocument document = PdfDocument(inputBytes: File(templatePath).readAsBytesSync());
+    for (int i = 0; i < document.form.fields.count; i++) {
+      final PdfField field = document.form.fields[i];
+      switch(i) {
+        case 0:
```

Comment:
Rather than using index please check the name of the field. Later if we add new field to pdf then the code needs to rewritten.